components/script/dom/request.rs

```diff
@@ -308,21 +308,26 @@ impl Request {
         // Step 27
         let mut headers_copy = r.Headers();
 
-        // This is equivalent to the specification's concept of
-        // "associated headers list".
-        if let RequestInfo::Request(ref input_request) = input {
-            headers_copy = input_request.Headers();
-        }
-
         // Step 28
```

Comment:
I'll leave them as it is for now! It's probably more straightforward to fix them when we update Request Constructor to match the spec :)